File: sample-code/php-5/software_version_numbers.php
<?php
require ("config.php");

# Where will the request be sent to
$url = 'https://api.whatismybrowser.com/api/v2/software_version_numbers/all';

# -- Set up HTTP Headers
$headers = [
    'X-API-KEY: '.$api_key,
];

# -- create a CURL handle containing the settings & data
$ch = curl_init();
curl_setopt($ch,CURLOPT_URL, $url);
curl_setopt($ch,CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);

# -- Make the request
$result = curl_exec($ch);
$curl_info = curl_getinfo($ch);
curl_close($ch);

# -- Try to decode the api response as json
$result_json = json_decode($result);
if ($result_json === null) {
    echo "Couldn't decode the response as JSON\n";
    exit();
}

# -- Check that the server responded with a "200/Success" code
if ($curl_info['http_code'] != 200) {
    echo "Didn't receive a 200 Success response from the API\n";
    echo "Instead, there was a ".$curl_info['http_code']." code\n";
    echo "The message was: ".$result_json->result->message."\n";
    exit();
}

# -- Check the API request was successful
if ($result_json->result->code != "success") {
    throw new Exception("The API did not return a 'success' response. It said: result: ".$result_json->result.", message_code: ".$result_json->message_code.", message: ".$result_json->message_code);
    exit();
}

# Now you have "$result_json" and can store, display or process any part of the response.

# -- print the entire json dump for reference
var_dump($result_json);

$version_data = $result_json->version_data;

foreach ($version_data as $software_key => $software_version_data) {

    echo "Version data for ".$software_key."\n";

    foreach ($software_version_data as $stream_code_key => $stream_version_data) {

        #var_dump($stream_version_data);

        echo "  Stream: ".$stream_code_key."\n";

        echo "\tThe latest version number for ".$software_key." [".$stream_code_key."] is ".join(".", $stream_version_data->latest_version)."\n";

        if ($stream_version_data->update) {
            echo "\tUpdate no: ".$stream_version_data->update."\n";
        }

        if ($stream_version_data->update_url) {
            echo "\tUpdate URL: ".$stream_version_data->update_url."\n";
        }

        if ($stream_version_data->download_url) {
            echo "\tDownload URL: ".$stream_version_data->download_url."\n";
        }
    }

    echo "---------------------------------\n";
}

File: sample-code/python-3.6/software_version_numbers.py
import requests
import json
import config

# Where will the request be sent to
api_url = "https://api.whatismybrowser.com/api/v2/software_version_numbers/all"

# -- Set up HTTP Headers
headers = {
    'X-API-KEY': config.api_key,
}

# -- Make the request
result = requests.get(api_url, headers=headers)

# -- Try to decode the api response as json
result_json = {}
try:
    result_json = result.json()
except Exception as e:
    print(result.text())
    print("Couldn't decode the response as JSON:", e)
    exit()

# -- Check that the server responded with a "200/Success" code
if result.status_code != 200:
    print("ERROR: not a 200 result. instead got: %s." % result.status_code)
    print(json.dumps(result_json, indent=2))
    exit()

# -- Check the API request was successful
if result_json.get('result', {}).get('code') != "success":
    print("The API did not return a 'success' response. It said: result code: %s, message_code: %s, message: %s" % (result_json.get('result', {}).get('code'), result_json.get('result', {}).get('message_code'), result_json.get('result', {}).get('message')))
    #print(json.dumps(result_json, indent=2))
    exit() 

# Now you have "result_json" and can store, display or process any part of the response.

# -- Print the entire json dump for reference
print(json.dumps(result_json, indent=2))

# -- Copy the `version_data` data to a variable for easier use
version_data = result_json.get('version_data')

# -- Loop over all the different software version data elements
for software_key in version_data:

    print("Version data for %s" % software_key)

    software_version_data = version_data.get(software_key)

    for stream_code_key in software_version_data:

        #print(software_version_data.get(stream_code_key))

        print("  Stream: %s" % stream_code_key)

        print("\tThe latest version number for %s [%s] is %s" % (software_key, stream_code_key, ".".join(software_version_data.get(stream_code_key).get("latest_version"))))
        
        if software_version_data.get(stream_code_key).get("update"):
            print("\tUpdate no: %s" % software_version_data.get(stream_code_key).get("update"))

        if software_version_data.get(stream_code_key).get("update_url"):
            print("\tUpdate URL: %s" % software_version_data.get(stream_code_key).get("update_url"))

        if software_version_data.get(stream_code_key).get("download_url"):
            print("\tDownload URL: %s" % software_version_data.get(stream_code_key).get("download_url"))

    print("---------------------------------")
